pytket/pytket/qasm/qasm.py

```diff
@@ -854,35 +854,52 @@ def _get_optype_and_params(op: Op) -> Tuple[OpType, Optional[List[float]]]:
     return (optype, params)
 
 
+def hqs_header(header: str) -> bool:
+    if header == "hqslib1":
+        return True
+
+    if header == "hqslib1_dev":
+        return True
+
+    return False
+
+
 def circuit_to_qasm_io(
```

Comment:
I guess we will need to implement the reverse translation too at some point -- is that left for a future PR?
It would probably require having a grammar specifying the allowed "special" operations. This could be hard-coded or -- for greater flexibility -- passed as a parameter.

Reply:
From my understanding that require to show all the possible options which we currently don't want to do?

Reply:
Not if the grammar was passed by the user as a parameter. Another option could be to interpret _all_ unparsed lines as barrier data (when the header is `hqslib1_dev`). I think we will need to implement this somehow, but we can leave it for now pending further discussion.